kernel/kernel-api/src/main/java/io/delta/kernel/internal/DeltaErrors.java

```diff
@@ -130,6 +134,41 @@ public static UnsupportedOperationException unsupportedWriteFeature(
                 writeProtocolVersion, writeFeature));
     }
 
+    public static IllegalArgumentException unsupportedDataType(DataType dataType) {
+        return new IllegalArgumentException("Unsupported data type: " + dataType);
```

Comment:
Can this maybe be "Kernel does not support writing data type %s" or something just to be clear (since this might be different from on the read side?